extensions/pkg/controller/utils.go

```diff
@@ -47,6 +49,12 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+const (
+	// ReferencedResourcesPrefix is the prefix used when copying referenced resources to the Shoot namespace in the Seed,
+	// to avoid naming collisions with resources managed by Gardener.
+	ReferencedResourcesPrefix = "ref-"
```

Comment:
I think, this const should rather be in `pkg/operation/botanist` or somewhere under `pkg`.

Reply:
Could not move the secret unless the above issue is solved, since this would lead to circular references.

Reply:
Done after getting rid of the dependency, see above.